homeassistant/components/govee_light_local/network.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+from ipaddress import IPv4Address, IPv6Address
+
+from homeassistant.components import network
+from homeassistant.core import HomeAssistant
+
+
+async def _async_get_all_source_ipv4_ips(
```

Comment:
I don't think we should re-invent the wheel here. The Network integration has `async_get_enabled_source_ips` which can easily be filtered down to IPv4 addresses.

Reply:
@mill1000 the problem with async_get_enabled_source_ips is that it doesnt actually get all the IPs. There is some logic that decideds which NICs are enabled or not based on like mDNS hops or something. at least for me it was calculating my secondary nic as disabled. so i had to write a function that didn't filter out "disabled" devices.

Reply:
Oh. Is your secondary NIC configured via System > Network?

Reply:
it shows up in system network in the UI if that is what you are asking.

![image](https://github.com/user-attachments/assets/5efacd01-3c40-4fd5-ba0c-7cfa303bde42)

Reply:
Hm.. that's quite a bit different looking than mine. Must be a HAOS vs container thing.

Here's mine where I manually enable a 2nd adapter. But I'm broadly assuming this is what "enables" another interface.
![image](https://github.com/user-attachments/assets/d97e875d-cd55-4e57-93c2-bf48205c822f)

Reply:
im running on supervised installer. i was looking for a UI like that....

Reply:
just to make sure i wasn't crazy i tested using the async_get_enabled_source_ips and it doesn't return the IP for my secondary NIC.

so idk if this is just a me problem, and i dont know how else i could work around it, or if some other problem.